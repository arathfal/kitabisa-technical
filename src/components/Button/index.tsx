import { ButtonHTMLAttributes } from 'react'

import classNames from '@/lib/classNames'
import { CommonProps } from '@/types/common'

type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> &
  CommonProps & {
    variant?: 'green' | 'red' | 'light'
  }

export default function Button({ className, disabled, children, ...props }: ButtonProps) {
  return (
    <button
      className={classNames(
        'h-10 w-auto rounded-md border border-slate-300 bg-white px-4 py-2 text-slate-950 transition-all hover:bg-slate-300',
        disabled && 'pointer-events-none bg-slate-500 text-white',
        className
      )}
      disabled={disabled}
      {...props}
    >
      {children}
    </button>
  )
}
