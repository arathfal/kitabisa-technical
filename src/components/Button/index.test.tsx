import { cleanup, render } from '@testing-library/react'

import Button from '.'

const renderButton = ({ className = '', text = '' }) => ({
  ...render(
    <Button className={className} data-testid="button-test">
      {text}
    </Button>
  )
})

describe('Button Component', () => {
  afterEach(cleanup)

  test('should render with text "Button Test"', () => {
    const { asFragment } = renderButton({
      text: 'Button Test'
    })
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render with class contain "bg-pink-400"', () => {
    const { asFragment } = renderButton({
      text: 'Button Test',
      className: 'bg-pink-400'
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
