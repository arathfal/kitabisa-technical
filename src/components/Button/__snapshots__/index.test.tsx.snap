// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render with class contain "bg-pink-400" 1`] = `
<DocumentFragment>
  <button
    class="w-auto rounded-md border border-slate-300 px-4 py-2 leading-none text-slate-950 transition-all hover:bg-slate-300 bg-pink-400"
    data-testid="button-test"
  >
    Button Test
  </button>
</DocumentFragment>
`;

exports[`Button Component should render with text "Button Test" 1`] = `
<DocumentFragment>
  <button
    class="w-auto rounded-md border border-slate-300 bg-white px-4 py-2 leading-none text-slate-950 transition-all hover:bg-slate-300"
    data-testid="button-test"
  >
    Button Test
  </button>
</DocumentFragment>
`;
